src/analysis/convex/simplicial_complex/basic.lean

```diff
@@ -0,0 +1,188 @@
+  inter_subset_convex_hull := λ s t hs ht, K.inter_subset_convex_hull (subset hs) (subset ht) }
+
+variables {𝕜 E}
+
+lemma convex_hull_face_subset (hs : s ∈ K.faces) : convex_hull 𝕜 ↑s ⊆ (K : set E) :=
+subset_bUnion_of_mem hs
+
+protected lemma subset (hs : s ∈ K.faces) : (s : set E) ⊆ K :=
+(subset_convex_hull 𝕜 _).trans (convex_hull_face_subset hs)
+
+lemma convex_hull_inter_convex_hull (hs : s ∈ K.faces) (ht : t ∈ K.faces) :
+  convex_hull 𝕜 ↑s ∩ convex_hull 𝕜 ↑t = convex_hull 𝕜 (s ∩ t : set E) :=
+(K.inter_subset_convex_hull hs ht).antisymm $ subset_inter
+  (convex_hull_mono $ set.inter_subset_left _ _) $ convex_hull_mono $ set.inter_subset_right _ _
+
+/-! ### Vertices -/
+
+/-- The vertices of a simplicial complex are its zero dimensional faces. -/
+def vertices (K : simplicial_complex 𝕜 E) : set E := {x | {x} ∈ K.faces}
+
+lemma mem_vertices : x ∈ K.vertices ↔ {x} ∈ K.faces := by refl
+
+lemma vertices_eq : K.vertices = ⋃ k ∈ K.faces, (k : set E) :=
+begin
+  ext x,
+  refine ⟨λ h, mem_bUnion h $ mem_coe.2 $ mem_singleton_self x, λ h, _⟩,
+  obtain ⟨s, hs, hx⟩ := mem_bUnion_iff.1 h,
+  exact K.down_closed hs (finset.singleton_subset_iff.2 $ mem_coe.1 hx) (singleton_ne_empty _),
+end
+
+lemma vertices_subset : K.vertices ⊆ K :=
+vertices_eq.subset.trans $ set.bUnion_mono $ λ x hx, subset_convex_hull 𝕜 x
+
+lemma vertex_mem_convex_hull_iff (hx : x ∈ K.vertices) (hs : s ∈ K.faces) :
+  x ∈ convex_hull 𝕜 (s : set E) ↔ x ∈ s :=
+begin
+  refine ⟨λ h, _, λ h, subset_convex_hull _ _ h⟩,
+  classical,
+  have h := K.inter_subset_convex_hull hx hs ⟨by simp, h⟩,
+  by_contra H,
+  rwa [←coe_inter, finset.disjoint_iff_inter_eq_empty.1
+    (finset.disjoint_singleton_right.2 H).symm, coe_empty, convex_hull_empty] at h,
+end
+
+/-- A face is a subset of another one iff its vertices are.  -/
+lemma face_subset_face_iff (hs : s ∈ K.faces) (ht : t ∈ K.faces) :
+  convex_hull 𝕜 (s : set E) ⊆ convex_hull 𝕜 ↑t ↔ s ⊆ t :=
+⟨λ h x hxs, (vertex_mem_convex_hull_iff (K.down_closed hs (finset.singleton_subset_iff.2 hxs) $
+  singleton_ne_empty _) ht).1 (h (subset_convex_hull 𝕜 ↑s hxs)), convex_hull_mono⟩
+
+/-! ### Facets -/
+
+/-- A facet of a simplicial complex is a maximal face. -/
+def facets (K : simplicial_complex 𝕜 E) : set (finset E) :=
+{s ∈ K.faces | ∀ ⦃t⦄, t ∈ K.faces → s ⊆ t → s = t}
+
+lemma mem_facets : s ∈ K.facets ↔ s ∈ K.faces ∧ ∀ t ∈ K.faces, s ⊆ t → s = t := mem_sep_iff
+
+lemma facets_subset : K.facets ⊆ K.faces := λ s hs, hs.1
+
+lemma not_facet_iff_subface (hs : s ∈ K.faces) : (s ∉ K.facets ↔ ∃ t, t ∈ K.faces ∧ s ⊂ t) :=
+begin
+  refine ⟨λ (hs' : ¬ (_ ∧ _)), _, _⟩,
+  { push_neg at hs',
+    obtain ⟨t, ht⟩ := hs' hs,
+    exact ⟨t, ht.1, ⟨ht.2.1, (λ hts, ht.2.2 (subset.antisymm ht.2.1 hts))⟩⟩ },
+  { rintro ⟨t, ht⟩ ⟨hs, hs'⟩,
+    have := hs' ht.1 ht.2.1,
+    rw this at ht,
+    exact ht.2.2 (subset.refl t) } -- `has_ssubset.ssubset.ne` would be handy here
+end
+
+/-! ### The semilattice of simplicial complexes -/
```

Comment:
Maybe explain in words in this docstring the meaning of `K <= L`--right now even the definition is somewhere hard to spot.